### PR TITLE
Add FreeMarker macro support in spring-webflux

### LINKED ISSUE #22950: spring.ftl is missing in spring-webflux
**Affects:** `spring-webflux` 5.1.6

In `org.springframework.web.reactive.result.view.freemarker.FreeMarkerConfigurer` the `spring.ftl` is mentioned, but actually does not exist.

I tried to use the one shipped with `spring-webmc`, but that led to further problems.

Please provide a webflux ready version of the ftl. 

Thank you!

### PR DESCRIPTION
**Pull request by ielatif**
See #22950 for details.

### CODE REVIEW

File: spring-webflux/src/main/java/org/springframework/web/reactive/result/view/freemarker/FreeMarkerConfigurer.java [hunk 1]
```diff
@@ -59,6 +57,7 @@
  * Note: Spring's FreeMarker support requires FreeMarker 2.3 or higher.
  *
  * @author Rossen Stoyanchev
+ * @author Issam El-atif
```
Comment (sbrannen) [inline spring-webflux/src/main/java/org/springframework/web/reactive/result/view/freemarker/FreeMarkerConfigurer.java]:
```suggestion
```

We don't add ourselves as authors of a class if the only change we make is deletion of a TODO. So please revert this change.

File: spring-webflux/src/main/resources/org/springframework/web/reactive/result/view/freemarker/spring.ftl [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+ * Spring's FreeMarker support will automatically make this file and therefore
+ * all macros within it available to any application using Spring's
+ * FreeMarkerConfigurer.
+ *
+ * To take advantage of these macros, the "exposeSpringMacroHelpers" property
+ * of the FreeMarker class needs to be set to "true". This will expose a
+ * RequestContext under the name "springMacroRequestContext", as needed by
+ * the macros in this library.
+ *
+ * @author Issam El-atif
```
Comment (sbrannen) [inline spring-webflux/src/main/resources/org/springframework/web/reactive/result/view/freemarker/spring.ftl]:
```suggestion
 * @author Darren Davison
 * @author Juergen Hoeller
 * @author Issam El-atif
```

Please include the original authors.

**Review by sbrannen — CHANGES_REQUESTED** [review]
I've requested two minor changes.

Please introduce tests for this functionality as well.

Thanks in advance!

File: spring-webflux/src/main/java/org/springframework/web/reactive/result/view/freemarker/FreeMarkerConfigurer.java (hunk 1 above)
Comment (ielatif) [reply spring-webflux/src/main/java/org/springframework/web/reactive/result/view/freemarker/FreeMarkerConfigurer.java]: Done. Got it :)

File: spring-webflux/src/main/resources/org/springframework/web/reactive/result/view/freemarker/spring.ftl (hunk 2 above)
Comment (ielatif) [reply spring-webflux/src/main/resources/org/springframework/web/reactive/result/view/freemarker/spring.ftl]: Done.

**Comment by ielatif** [comment]
> Please introduce tests for this functionality as well

Done 👍

**Review by sbrannen — APPROVED** [review]

**Review by sbrannen — CHANGES_REQUESTED** [review]
Thanks for making the recent changes and adding the tests!

Unfortunately, we have recently switched to AssertJ for all assertions. So the test classes will fail the build due to Checkstyle violations.

Please update the tests to use AssertJ instead of Hamcrest and JUnit 4.

You can use the latest versions (in `master`) of the test classes you copied as a basis.

Thanks in advance.

**Comment by ielatif** [comment]
@sbrannen Thanks.
Migrated JUnit 4 assertions to AssertJ

**Review by sbrannen — APPROVED** [review]

**Comment by sbrannen** [comment]
This has been merged into `master` in b58f34d802f4503789bbf94576e1cb9c8eb92eb8 and refined in 2f640fe20569a6e08794934c579e3ad00550325a.

Thanks again for the PR!
